macaw-aarch32/src/Data/Macaw/ARM/Semantics/TH.hs

```diff
@@ -789,7 +789,7 @@ refField :: Ctx.Size ctx -> Ctx.Index ctx tp -> BoundExp -> MacawQ arch t fs Bou
 refField sz idx e = case Ctx.viewSize sz of
   Ctx.IncSize sz' | Ctx.ZeroSize <- Ctx.viewSize sz' -> return e
   _ -> case e of
-    EagerBoundExp (TupE es) | Ctx.indexVal idx < length es -> return $ EagerBoundExp $ es !! (Ctx.indexVal idx)
+    EagerBoundExp (TupE es) | Ctx.indexVal idx < length es -> return $ EagerBoundExp $ es !! (Ctx.indexVal idx) -- TODO: fix error
```

Comment:
```
[ 5 of 11] Compiling Data.Macaw.ARM.Semantics.TH ( src/Data/Macaw/ARM/Semantics/TH.hs, /project/macaw/dist-newstyle/build/x86_64-linux/ghc-8.10.1/macaw-aarch32-0.1.0.0/build/Data/Macaw/ARM/Semantics/TH.o, /project/macaw/dist-newstyle/build/x86_64-linux/ghc-8.10.1/macaw-aarch32-0.1.0.0/build/Data/Macaw/ARM/Semantics/TH.dyn_o )

src/Data/Macaw/ARM/Semantics/TH.hs:792:88: error:
    • Couldn't match expected type ‘Exp’ with actual type ‘Maybe Exp’
    • In the second argument of ‘($)’, namely
        ‘es !! (Ctx.indexVal idx)’
      In the second argument of ‘($)’, namely
        ‘EagerBoundExp $ es !! (Ctx.indexVal idx)’
      In the expression:
        return $ EagerBoundExp $ es !! (Ctx.indexVal idx)
    |
792 |     EagerBoundExp (TupE es) | Ctx.indexVal idx < length es -> return $ EagerBoundExp $ es !! (Ctx.indexVal idx)
    |        
```

Reply:
@danmatichuk  has a fix for this in #170, I believe

Reply:
Yup, that fixes it, so that needs to go in first.

Reply:
I believe that one is ready to merge